nocopy_linkbuffer.go

```diff
@@ -124,28 +129,50 @@ func (b *UnsafeLinkBuffer) Peek(n int) (p []byte, err error) {
 	if b.isSingleNode(n) {
 		return b.read.Peek(n), nil
 	}
+
 	// multiple nodes
-	var pIdx int
-	if block1k < n && n <= mallocMax {
-		p = malloc(n, n)
-		b.caches = append(b.caches, p)
-	} else {
-		p = dirtmake.Bytes(n, n)
-	}
-	var node = b.read
-	var l int
-	for ack := n; ack > 0; ack = ack - l {
-		l = node.Len()
-		if l >= ack {
-			pIdx += copy(p[pIdx:], node.Peek(ack))
-			break
-		} else if l > 0 {
-			pIdx += copy(p[pIdx:], node.Peek(l))
+	// always use malloc, since we will reuse b.cachePeek
+	if b.cachePeek != nil && cap(b.cachePeek) < n {
```

Comment:
here should notice that currently malloc(4095, 4095) may return a slice (len=4095, cap=4096).  But I see seems it not effect the logic bellow. Just for a infromation when using cap of a buffer from mcache

Reply:
yes, noted.